sessions/backend-create/products/README.md

```diff
@@ -0,0 +1,115 @@
+# Backend Create: Products
+
+In this challenge, you'll further expand a fish shop. This time, you'll create new data in your local MongoDB using `mongoose` and refresh the UI programmatically to display the new product immediately.
+
+> 💡 You can use this template as a starting point. But if you are far enough along with your own Fish Shop App, please use that instead.
+
+## Task
+
+### Prepare your Database
+
+If you have not done so, use MongoDB Compass to create a database:
+
+- the database should be called `fish-shop`,
+- there should be two collections: `products` and `reviews`,
+- download and extract the [resources](README.md#resources) and
+- import the data of the `products.json` file into your `products` collection;
+- import the data of the `reviews.json` file into your `reviews` collection.
+
+Create a `.env.local` file based on the [`.env.local.example`](./.env.local.example). Be sure to spell the name of the database (`fish-shop`) correctly.
+
+### Introduction
+
+Run `npm run dev` and open `localhost:3000` in your browser.
+
+Have a look around:
+
+- there is an overview page with a form to add a new fish and a list of all products below that form;
+- when clicking a product in the list, you'll be redirected to a details page;
+- note that submitting the form does not do anything right now.
+
+Your task is to refactor the app so that submitting the form creates a new entry in your local MongoDB and refreshes the page to show all products (including the new one).
+
+### Add a `POST` route
+
+Switch to [`pages/api/products/index.js`](./pages/api/products/index.js) and write the code for the `request.method` `POST` :
+
+- Use a `try...catch` block.
+- Try to:
+  - Save the product data submitted by the form - accessible in `request.body` - to a variable called `productData`.
+  - Pass the `productData` a `new Product(data_goes_here)` and save it in a variable called `product`.
+  - _Wait_ until the new product was saved in the database with `product.save()`.
+  - Respond with a status `201` and the message `{ status: "Product created." }`.
+- If an error occurs:
+  - Log the error to the console.
+  - Respond with a status `400` and the message `{ error: error.message }`.
+
+Submitting the form will not yet work because the form does not know that you've created a `POST` route it can use.
+
+### Send a `POST` request
+
+Switch to [`components/ProductForm/index.js`](./components/ProductForm/index.js):
+
+- There already is an `onSubmit` function which creates a `productData` object with all relevant data.
```

Comment:
AFAIK (and according to Alex) at the point of 'initiation' the handler function should be prefixed with "on", such as "onClick", "onSubmit", etc. Something along the lines of `onSubmit={onSubmit}`.

Reply:
Well, I've checked the entire `web-exercises` repo: I think @JessicaLoers is right here, because we have always initiated the handler function as `handleSubmit`, but passed it as `onSubmit` to components (or a HTML form element, as in this case).
I'll change that.